datafusion/core/src/datasource/physical_plan/arrow_file.rs

```diff
@@ -311,7 +311,7 @@ impl FileOpener for ArrowOpener {
                     {
                         decoder.read_dictionary(
                             dict_block,
-                            &Buffer::from_bytes(dict_result.into()),
+                            &Buffer::from(dict_result),
```

Comment:
- Due to https://github.com/apache/arrow-rs/pull/6939 (make it easier to convert from `Bytes` to `Buffer`)